csharp/extractor/Semmle.Extraction.CIL/Entities/SignatureDecoder.cs

```diff
@@ -138,21 +139,28 @@ public void WriteId(TextWriter trapFile, GenericContext gc)
         private class Modified : ITypeSignature
         {
             private readonly ITypeSignature unmodifiedType;
+            private readonly ITypeSignature modifier;
+            private readonly bool isRequired;
 
-            public Modified(ITypeSignature unmodifiedType)
+            public Modified(ITypeSignature unmodifiedType, ITypeSignature modifier, bool isRequired)
             {
                 this.unmodifiedType = unmodifiedType;
+                this.modifier = modifier;
+                this.isRequired = isRequired;
             }
 
             public void WriteId(TextWriter trapFile, GenericContext gc)
             {
                 unmodifiedType.WriteId(trapFile, gc);
+                trapFile.Write(isRequired ? " modreq(" : " modopt(");
+                modifier.WriteId(trapFile, gc);
+                trapFile.Write(")");
```

Comment:
Why is this added?I thought modified types were not written to TRAP.

Reply:
Thanks for the detailed explanation; I am convinced! :-) (I hadn't realized that the IDs written here were only part of method signatures.)